test/built-ins/Temporal/PlainDateTime/prototype/withCalendar/argument-string.js

```diff
@@ -0,0 +1,23 @@
+// Copyright (C) 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.plaindatetime.prototype.withcalendar
+description: String argument, if it names a recognizable calendar, gets cast
+features: [Temporal]
+includes: [temporalHelpers.js]
+---*/
+
+const dt = new Temporal.PlainDateTime(1976, 11, 18, 15, 23, 30, 123, 456, 789);
```

Comment:
This test is OK as written as far as I'm concerned, but it could be made even better: construct `dt` with a custom plain-object calendar with a `toString` that returns something that's not `iso8601`, and then test after the call to `withCalendar("iso8601")` that the `calendar` property of the result is an instance of `Temporal.Calendar` whose `toString` returns `iso8601`.